src/Sensor/SensorOptical.cpp

```diff
@@ -52,3 +57,8 @@ float SensorOptical::getAngularVelocity() {
     float ret = ((float)n / NUM_APERTURES) * 2 * PI * (MEGA / (float)deltaT);
     return ret;
 }
+
+float SensorOptical::getLinearVelocity() {
+    float ret = VelocityFactor * getAngularVelocity();
```

Comment:
the fact that computing angular velocity modifies the state of SensorOptical is problematic, say if we want to record angular and linear velocity readings.  I made some comments in your other PR for the direction I was planning on taking this class